great_expectations/rule_based_profiler/helpers/util.py

```diff
@@ -1050,8 +1051,15 @@ def get_or_create_expectation_suite(
     return expectation_suite
 
 
-def sanitize_parameter_name(name: str) -> str:
+def sanitize_parameter_name(
+    name: str,
+    suffix: Optional[str] = None,
+) -> str:
     """
-    This method provides display-friendly version of "name" argument.
+    This method provides display-friendly version of "name" argument (with optional "suffix" argument).
     """
+    if suffix:
```

Comment:
I am unable to follow what purpose this `suffix` serves.

Reply:
@NathanFarmer The suffix enables us to make parameter names unique.  Without it, what can happen is that same metric name, same domain kwargs, but different value kwargs can cause two metric configurations to map to the same parameter name.  By adding a suffix, we can disambiguate them.  Thanks.

Reply:
@alexsherstinsky Why do we pass a `suffix` string if we are just going to then hash it? Also, how does `suffix` relate to value kwargs in the code itself? How is the user supposed to know what `suffix` means?

Reply:
@NathanFarmer Please see improved `docstring` and additional unit tests (plus some more "beautification").  Thanks!